src/simple_experiments/events/simple_card.cljs

```diff
@@ -25,23 +19,63 @@
     :pending-registration
     :associated})
 
-(defn set-active-card [{:keys [db]} [_ card-uuid status]]
-  (if (nil? card-uuid)
-    {}
-    (let [sdid (six-digit-id card-uuid)]
-      {:db (assoc-in db [:ui :active-card]
-                     {:uuid card-uuid
-                      :six-digit-id sdid
-                      :six-digit-display (six-digit-display sdid)
-                      :status status})})))
+(defn has-six-digit-id? [patient six-digit-id]
+  (let [six-digit-ids (set (concat (:six-digit-ids patient)
+                                   (map ->six-digit-id
+                                        (:card-uuids patient))))]
+    (contains? six-digit-ids six-digit-id)))
+
+(defn find-patients [db six-digit-id]
+  (->> db
+       :store
+       :patients
+       vals
+       (filter #(has-six-digit-id? % six-digit-id))))
+
+(defn handle-six-digit-keyboard [{:keys [db]} [_ six-digit-id]]
+  {:db (assoc-in db [:ui :simple-card :six-digit-id] six-digit-id)})
+
+(defn handle-six-digit-input [{:keys [db]} [_]]
+  (let [six-digit-id (get-in db [:ui :simple-card :six-digit-id])
+        existing-patients (find-patients db six-digit-id)]
+    (case (nav/previous-screen)
+      :home
+      (if (empty? existing-patients)
+        {:dispatch-n [[:goto :patient-list]
+                      [:patient-search-clear]
+                      [:goto-search-mode]
+                      [:set-active-card nil six-digit-id :pending-association]]}
+        {:db (assoc-in db [:ui :patient-search :results] existing-patients)
+         :dispatch-n [[:goto :patient-list]
+                      [:goto-select-mode]
+                      [:set-active-card nil six-digit-id :pending]]})
+
+      :new-patient
+      {:dispatch-n [[:set-active-card nil six-digit-id :pending-registration]
+                    [:go-back]]}
+
+      {})))
+
+(defn set-active-card [{:keys [db]} [_ card-uuid six-digit-id status]]
+  (let [sdid (or six-digit-id
+                 (->six-digit-id card-uuid))]
+    {:db (assoc-in db [:ui :active-card]
+                   {:uuid card-uuid
+                    :six-digit-id sdid
+                    :six-digit-display (six-digit-display sdid)
+                    :status status})}))
 
 (defn clear-active-card [{:keys [db]} _]
   {:db (assoc-in db [:ui :active-card] nil)})
 
-(defn associate-simple-card-with-patient [{:keys [db]} [_ card-uuid patient-id]]
-  {:db (-> db
-           (update-in [:store :patients patient-id :card-uuids] conj card-uuid)
-           (assoc-in [:ui :active-card :status] :associated))})
+(defn associate-simple-card-with-patient [{:keys [db]} [_ card patient-id]]
+  (let [uuid (:uuid card)
+        sdid (:six-digit-id card)
+        field-to-update (if uuid :card-uuids :six-digit-ids)
+        update-with-value (or uuid sdid)]
```

Comment:
`update-with-value` shouldn't be independent from the previous decision of the field to update. We should use something in the lines of:
```clojure
(case field-to-update
  :card-uuids uuid
  :six-digit-ids sdid)
```

Reply:
This is done in a couple of places, so we could extract this into a function too.

Reply:
agreed

Reply:
actually, i did try abstracting this out, and couldn't settle on the 'right' way to do this. let's pair on this.